language/for_spec.rb

```diff
@@ -114,6 +114,53 @@ class OFor
     $var = old_global_var
   end
 
+  it "allows an attribute as an iterator name" do
+    class OFor
+      attr_accessor :target
+    end
+
+    ofor = OFor.new
+    m = [1,2,3]
+    n = 0
+    for ofor.target in m
+      n += 1
+    end
+    ofor.target.should == 3
+    n.should == 3
+  end
+
+  ruby_version_is "3.4" do
```

Comment:
I would also add a link to he issue. I suppose the fix will be backported soon so the version guard will be removed.

Reply:
I added the link, but the bug report doesn't mention anything about backports, so I'm not sure if we should be expecting it. But the link doesn't hurt.

I added specs for the syntax `for a[0]/a[:a] in x` as well, according to Prism these are different parse nodes than the regular `for foo.bar in x`

Reply:
We should use `ruby_bug` for this, and it should be backported (segfault fixes should basically always be backported)